src/modules/esl-utils/dom/rect.ts

```diff
@@ -112,4 +127,11 @@ export class Rect {
   public shrink(decrementX: number, decrementY: number = decrementX): Rect {
     return this.grow(-decrementX, -decrementY);
   }
+
+  /**
+   * Method to check if element's area is bigger than zero
+   */
+  public isEmpty(): boolean {
+    return this.height <= 0 || this.width <= 0;
```

Comment:
Nope, it's OR condition on purpose, because there are cases when only one value is negative or equal to zero. Check test cases for example example
` test.each([
    [[0, 0, 10, 10], false],
    [[0, 0, -10, 10], true],
    [[0, 0, 10, -10], true],
    [[0, 0, 0, 0], true],
    [[0, 0, -10, -10], true]
  ])('isEmpty: %s', (coords, expected) => {
    expect(new Rect(...coords).isEmpty()).toBe(expected);
  });`

Reply:
Agree in that case that the emptiness of the rectangle is a really undefined term
is the point an empty rectangle - most likely, yes
is the line an empty rectangle - not sure 100%; depends on terminology or applicable task

To resolve the unclearness, propose moving to the square term.
In our case element is not visible when geometrically: the square of intersecting viewport and element rectangle is 0.

PS: negative values for width and height is not valid values that we will not see in reality, or in another point of view it is a valid inverse rectangle but it will not have a 0 square the same as it is not empty )

Reply:
The main reason of implementing isEmpty was to check if rectangle's area returned by Rect.intersect() is equal to zero or returned rectangle was sort of 'inverted'. Maybe, we can handle inversion differently, for example by getting max value between 0 and dimension values
```
const width = Math.max(right - x, 0);
const height = Math.max(bottom - y, 0);
```